cypress/integration/SelectLabs.spec.ts

```diff
@@ -337,6 +337,92 @@ describe('Select', () => {
     });
   });
 
+  context(`given the "Default" story is rendered`, () => {
+    beforeEach(() => {
+      h.stories.load('Labs|Select/React/Top Label', 'Default');
+    });
+
+    context('when the menu is opened', () => {
+      beforeEach(() => {
+        cy.findByLabelText('Label')
+          .focus()
+          .type('{downarrow}');
+      });
+
+      context('the menu', () => {
+        it('should set assistive focus to the first option ("E-mail")', () => {
+          cy.findByLabelText('Label')
+            .pipe(h.selectLabs.getMenu)
+            .pipe(getAssistiveFocus)
+            .should('have.text', 'E-mail');
+        });
+      });
+
+      context('when focus is advanced to the second option ("Phone")', () => {
+        beforeEach(() => {
+          cy.focused().type('{downarrow}');
+        });
+
+        context('the menu', () => {
+          it('should set assistive focus to the second option ("Phone")', () => {
+            cy.findByLabelText('Label')
+              .pipe(h.selectLabs.getMenu)
+              .pipe(getAssistiveFocus)
+              .should('have.text', 'Phone');
+          });
+        });
+
+        context(
+          'when the menu is closed WITHOUT selecting the newly focused option ("Phone")',
+          () => {
+            beforeEach(() => {
+              cy.focused().type('{esc}');
+            });
+
+            context('when the menu is re-opened AFTER it has fully closed', () => {
+              beforeEach(() => {
+                // Wait a short period of time to give the menu time to fully close
+                // (so we don't interrupt the menu's closing animation and cause it
+                // to re-open while it's in the middle of closing)
+                cy.wait(1000);
```

Comment:
Is it possible to instead explicitly wait for it to close instead? Something like:

```suggestion
                // Wait for menu to fully close before opening again
                cy.findByLabelText('Label')
                  .pipe(h.selectLabs.getMenu)
                  .should('not.exist')

```

Reply:
Nice, I didn't think about using an assertion like this to force Cypress to wait for the assertion to pass before proceeding with the test -- thanks for the tip 👍